// @n3oltd/karakoram.platforms.sdk.types/esm/index.js
//----------------------
// <auto-generated>
//     Generated using the NSwag toolchain v14.6.0.0 (NJsonSchema v11.5.0.0 (Newtonsoft.Json v13.0.0.0)) (http://NSwag.org)
// </auto-generated>
//----------------------
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
/* eslint-disable */
// ReSharper disable InconsistentNaming
var ConnectApiBase = /** @class */ (function () {
    function ConnectApiBase(apiHeaders) {
        var _this = this;
        this.transformOptions = function (options) {
            var _a, _b;
            var headers = __assign({}, options.headers);
            var subscriptionId = _this.apiHeaders.getSubscriptionId();
            if (subscriptionId) {
                headers['N3O-Subscription-Id'] = subscriptionId;
            }
            var bearerToken = (_b = (_a = _this.apiHeaders).getBearerToken) === null || _b === void 0 ? void 0 : _b.call(_a);
            if (bearerToken) {
                headers['Authorization'] = bearerToken;
            }
            options.headers = headers;
            return Promise.resolve(options);
        };
        this.apiHeaders = apiHeaders;
    }
    return ConnectApiBase;
}());
export { ConnectApiBase };
/** One of 'scheduledGiving', 'standard', 'telethon' */
export var CampaignType;
(function (CampaignType) {
    CampaignType["ScheduledGiving"] = "scheduledGiving";
    CampaignType["Standard"] = "standard";
    CampaignType["Telethon"] = "telethon";
})(CampaignType || (CampaignType = {}));
/** One of 'feedback', 'fund', 'sponsorship' */
export var DesignationType;
(function (DesignationType) {
    DesignationType["Feedback"] = "feedback";
    DesignationType["Fund"] = "fund";
    DesignationType["Sponsorship"] = "sponsorship";
})(DesignationType || (DesignationType = {}));
/** One of 'campaign', 'campaignsList', 'fundDimensionSearch' */
export var DonateMenuEntryType;
(function (DonateMenuEntryType) {
    DonateMenuEntryType["Campaign"] = "campaign";
    DonateMenuEntryType["CampaignsList"] = "campaignsList";
    DonateMenuEntryType["FundDimensionSearch"] = "fundDimensionSearch";
})(DonateMenuEntryType || (DonateMenuEntryType = {}));
/** One of 'donateButton', 'donationForm' */
export var ElementType;
(function (ElementType) {
    ElementType["DonateButton"] = "donateButton";
    ElementType["DonationForm"] = "donationForm";
})(ElementType || (ElementType = {}));
/** One of 'oneTime', 'recurring' */
export var GiftType;
(function (GiftType) {
    GiftType["OneTime"] = "oneTime";
    GiftType["Recurring"] = "recurring";
})(GiftType || (GiftType = {}));
/** One of 'campaignsIndex', 'designationsIndex', 'donateMenu', 'platformsSearch' */
export var PlatformsSubscriptionFile;
(function (PlatformsSubscriptionFile) {
    PlatformsSubscriptionFile["CampaignsIndex"] = "campaignsIndex";
    PlatformsSubscriptionFile["DesignationsIndex"] = "designationsIndex";
    PlatformsSubscriptionFile["DonateMenu"] = "donateMenu";
    PlatformsSubscriptionFile["PlatformsSearch"] = "platformsSearch";
})(PlatformsSubscriptionFile || (PlatformsSubscriptionFile = {}));
