// @n3oltd/karakoram.checkout.sdk.network-merchants/esm/index.js
//----------------------
// <auto-generated>
//     Generated using the NSwag toolchain v14.6.0.0 (NJsonSchema v11.5.0.0 (Newtonsoft.Json v13.0.0.0)) (http://NSwag.org)
// </auto-generated>
//----------------------
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
/* eslint-disable */
// ReSharper disable InconsistentNaming
var ConnectApiBase = /** @class */ (function () {
    function ConnectApiBase(apiHeaders) {
        var _this = this;
        this.transformOptions = function (options) {
            var _a, _b;
            var headers = __assign({}, options.headers);
            var subscriptionId = _this.apiHeaders.getSubscriptionId();
            if (subscriptionId) {
                headers['N3O-Subscription-Id'] = subscriptionId;
            }
            var bearerToken = (_b = (_a = _this.apiHeaders).getBearerToken) === null || _b === void 0 ? void 0 : _b.call(_a);
            if (bearerToken) {
                headers['Authorization'] = bearerToken;
            }
            options.headers = headers;
            return Promise.resolve(options);
        };
        this.apiHeaders = apiHeaders;
    }
    return ConnectApiBase;
}());
export { ConnectApiBase };
var NetworkMerchantsClient = /** @class */ (function (_super) {
    __extends(NetworkMerchantsClient, _super);
    function NetworkMerchantsClient(configuration, baseUrl, http) {
        var _this = _super.call(this, configuration) || this;
        _this.jsonParseReviver = undefined;
        _this.http = http ? http : window;
        _this.baseUrl = baseUrl !== null && baseUrl !== void 0 ? baseUrl : "/";
        return _this;
    }
    /**
     * @param body (optional)
     * @return OK
     */
    NetworkMerchantsClient.prototype.processCredential = function (methodId, flowId, scopeId, body) {
        var _this = this;
        var url_ = this.baseUrl + "/networkmerchants/v1.0/credentials/{methodId}/{flowId}/{scopeId}/process";
        if (methodId === undefined || methodId === null)
            throw new globalThis.Error("The parameter 'methodId' must be defined.");
        url_ = url_.replace("{methodId}", encodeURIComponent("" + methodId));
        if (flowId === undefined || flowId === null)
            throw new globalThis.Error("The parameter 'flowId' must be defined.");
        url_ = url_.replace("{flowId}", encodeURIComponent("" + flowId));
        if (scopeId === undefined || scopeId === null)
            throw new globalThis.Error("The parameter 'scopeId' must be defined.");
        url_ = url_.replace("{scopeId}", encodeURIComponent("" + scopeId));
        url_ = url_.replace(/[?&]$/, "");
        var content_ = JSON.stringify(body);
        var options_ = {
            body: content_,
            method: "POST",
            headers: {
                "Content-Type": "application/json",
                "Accept": "application/json"
            }
        };
        return this.transformOptions(options_).then(function (transformedOptions_) {
            return _this.http.fetch(url_, transformedOptions_);
        }).then(function (_response) {
            return _this.processProcessCredential(_response);
        });
    };
    NetworkMerchantsClient.prototype.processProcessCredential = function (response) {
        var _this = this;
        var status = response.status;
        var _headers = {};
        if (response.headers && response.headers.forEach) {
            response.headers.forEach(function (v, k) { return _headers[k] = v; });
        }
        ;
        if (status === 200) {
            return response.text().then(function (_responseText) {
                var result200 = null;
                result200 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return result200;
            });
        }
        else if (status === 204) {
            return response.text().then(function (_responseText) {
                return throwException("No Content", status, _responseText, _headers);
            });
        }
        else if (status === 400) {
            return response.text().then(function (_responseText) {
                var result400 = null;
                result400 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return throwException("Bad Request", status, _responseText, _headers, result400);
            });
        }
        else if (status === 401) {
            return response.text().then(function (_responseText) {
                var result401 = null;
                result401 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return throwException("Unauthorized", status, _responseText, _headers, result401);
            });
        }
        else if (status === 404) {
            return response.text().then(function (_responseText) {
                var result404 = null;
                result404 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return throwException("Not Found", status, _responseText, _headers, result404);
            });
        }
        else if (status === 412) {
            return response.text().then(function (_responseText) {
                var result412 = null;
                result412 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return throwException("Precondition Failed", status, _responseText, _headers, result412);
            });
        }
        else if (status === 500) {
            return response.text().then(function (_responseText) {
                return throwException("Internal Server Error", status, _responseText, _headers);
            });
        }
        else if (status !== 200 && status !== 204) {
            return response.text().then(function (_responseText) {
                return throwException("An unexpected server error occurred.", status, _responseText, _headers);
            });
        }
        return Promise.resolve(null);
    };
    /**
     * @param body (optional)
     * @return OK
     */
    NetworkMerchantsClient.prototype.processPayment = function (methodId, flowId, scopeId, body) {
        var _this = this;
        var url_ = this.baseUrl + "/networkmerchants/v1.0/payments/{methodId}/{flowId}/{scopeId}/process";
        if (methodId === undefined || methodId === null)
            throw new globalThis.Error("The parameter 'methodId' must be defined.");
        url_ = url_.replace("{methodId}", encodeURIComponent("" + methodId));
        if (flowId === undefined || flowId === null)
            throw new globalThis.Error("The parameter 'flowId' must be defined.");
        url_ = url_.replace("{flowId}", encodeURIComponent("" + flowId));
        if (scopeId === undefined || scopeId === null)
            throw new globalThis.Error("The parameter 'scopeId' must be defined.");
        url_ = url_.replace("{scopeId}", encodeURIComponent("" + scopeId));
        url_ = url_.replace(/[?&]$/, "");
        var content_ = JSON.stringify(body);
        var options_ = {
            body: content_,
            method: "POST",
            headers: {
                "Content-Type": "application/json",
                "Accept": "application/json"
            }
        };
        return this.transformOptions(options_).then(function (transformedOptions_) {
            return _this.http.fetch(url_, transformedOptions_);
        }).then(function (_response) {
            return _this.processProcessPayment(_response);
        });
    };
    NetworkMerchantsClient.prototype.processProcessPayment = function (response) {
        var _this = this;
        var status = response.status;
        var _headers = {};
        if (response.headers && response.headers.forEach) {
            response.headers.forEach(function (v, k) { return _headers[k] = v; });
        }
        ;
        if (status === 200) {
            return response.text().then(function (_responseText) {
                var result200 = null;
                result200 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return result200;
            });
        }
        else if (status === 204) {
            return response.text().then(function (_responseText) {
                return throwException("No Content", status, _responseText, _headers);
            });
        }
        else if (status === 400) {
            return response.text().then(function (_responseText) {
                var result400 = null;
                result400 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return throwException("Bad Request", status, _responseText, _headers, result400);
            });
        }
        else if (status === 401) {
            return response.text().then(function (_responseText) {
                var result401 = null;
                result401 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return throwException("Unauthorized", status, _responseText, _headers, result401);
            });
        }
        else if (status === 404) {
            return response.text().then(function (_responseText) {
                var result404 = null;
                result404 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return throwException("Not Found", status, _responseText, _headers, result404);
            });
        }
        else if (status === 412) {
            return response.text().then(function (_responseText) {
                var result412 = null;
                result412 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return throwException("Precondition Failed", status, _responseText, _headers, result412);
            });
        }
        else if (status === 500) {
            return response.text().then(function (_responseText) {
                return throwException("Internal Server Error", status, _responseText, _headers);
            });
        }
        else if (status !== 200 && status !== 204) {
            return response.text().then(function (_responseText) {
                return throwException("An unexpected server error occurred.", status, _responseText, _headers);
            });
        }
        return Promise.resolve(null);
    };
    return NetworkMerchantsClient;
}(ConnectApiBase));
export { NetworkMerchantsClient };
/** One of 'AD', 'AE', 'AF', 'AG', 'AI', 'AL', 'AM', 'AN', 'AO', 'AR', 'AS', 'AT', 'AU', 'AW', 'AX', 'AZ', 'BA', 'BB', 'BD', 'BE', 'BF', 'BG', 'BH', 'BI', 'BJ', 'BL', 'BM', 'BN', 'BO', 'BQ', 'BR', 'BS', 'BT', 'BW', 'BY', 'BZ', 'CA', 'CC', 'CD', 'CF', 'CG', 'CH', 'CI', 'CK', 'CL', 'CM', 'CN', 'CO', 'CR', 'CU', 'CV', 'CW', 'CX', 'CY', 'CZ', 'DE', 'DJ', 'DK', 'DM', 'DO', 'DZ', 'EC', 'EE', 'EG', 'EH', 'ER', 'ES', 'ET', 'FI', 'FJ', 'FK', 'FM', 'FO', 'FR', 'GA', 'GB', 'GD', 'GE', 'GF', 'GG', 'GH', 'GI', 'GL', 'GM', 'GN', 'GP', 'GQ', 'GR', 'GS', 'GT', 'GU', 'GW', 'GY', 'HK', 'HN', 'HR', 'HT', 'HU', 'ID', 'IE', 'IL', 'IM', 'IN', 'IO', 'IQ', 'IR', 'IS', 'IT', 'JE', 'JM', 'JO', 'JP', 'KE', 'KG', 'KH', 'KI', 'KM', 'KN', 'KP', 'KR', 'KW', 'KY', 'KZ', 'LA', 'LB', 'LC', 'LI', 'LK', 'LR', 'LS', 'LT', 'LU', 'LV', 'LY', 'MA', 'MC', 'MD', 'ME', 'MF', 'MG', 'MH', 'MK', 'ML', 'MM', 'MN', 'MO', 'MP', 'MQ', 'MR', 'MS', 'MT', 'MU', 'MV', 'MW', 'MX', 'MY', 'MZ', 'NA', 'NC', 'NE', 'NF', 'NG', 'NI', 'NL', 'NO', 'NP', 'NR', 'NU', 'NZ', 'OM', 'PA', 'PE', 'PF', 'PG', 'PH', 'PK', 'PL', 'PM', 'PN', 'PR', 'PS', 'PT', 'PW', 'PY', 'QA', 'RE', 'RO', 'RS', 'RU', 'RW', 'SA', 'SB', 'SC', 'SD', 'SE', 'SG', 'SH', 'SI', 'SJ', 'SK', 'SL', 'SM', 'SN', 'SO', 'SR', 'SS', 'ST', 'SV', 'SX', 'SY', 'SZ', 'TC', 'TD', 'TG', 'TH', 'TJ', 'TK', 'TL', 'TM', 'TN', 'TO', 'TR', 'TT', 'TV', 'TW', 'TZ', 'UA', 'UG', 'UM', 'US', 'UY', 'UZ', 'VA', 'VC', 'VE', 'VG', 'VI', 'VN', 'VU', 'WF', 'WS', 'XK', 'YE', 'YT', 'ZA', 'ZM', 'ZW' */
export var Country;
(function (Country) {
    Country["AD"] = "AD";
    Country["AE"] = "AE";
    Country["AF"] = "AF";
    Country["AG"] = "AG";
    Country["AI"] = "AI";
    Country["AL"] = "AL";
    Country["AM"] = "AM";
    Country["AN"] = "AN";
    Country["AO"] = "AO";
    Country["AR"] = "AR";
    Country["AS"] = "AS";
    Country["AT"] = "AT";
    Country["AU"] = "AU";
    Country["AW"] = "AW";
    Country["AX"] = "AX";
    Country["AZ"] = "AZ";
    Country["BA"] = "BA";
    Country["BB"] = "BB";
    Country["BD"] = "BD";
    Country["BE"] = "BE";
    Country["BF"] = "BF";
    Country["BG"] = "BG";
    Country["BH"] = "BH";
    Country["BI"] = "BI";
    Country["BJ"] = "BJ";
    Country["BL"] = "BL";
    Country["BM"] = "BM";
    Country["BN"] = "BN";
    Country["BO"] = "BO";
    Country["BQ"] = "BQ";
    Country["BR"] = "BR";
    Country["BS"] = "BS";
    Country["BT"] = "BT";
    Country["BW"] = "BW";
    Country["BY"] = "BY";
    Country["BZ"] = "BZ";
    Country["CA"] = "CA";
    Country["CC"] = "CC";
    Country["CD"] = "CD";
    Country["CF"] = "CF";
    Country["CG"] = "CG";
    Country["CH"] = "CH";
    Country["CI"] = "CI";
    Country["CK"] = "CK";
    Country["CL"] = "CL";
    Country["CM"] = "CM";
    Country["CN"] = "CN";
    Country["CO"] = "CO";
    Country["CR"] = "CR";
    Country["CU"] = "CU";
    Country["CV"] = "CV";
    Country["CW"] = "CW";
    Country["CX"] = "CX";
    Country["CY"] = "CY";
    Country["CZ"] = "CZ";
    Country["DE"] = "DE";
    Country["DJ"] = "DJ";
    Country["DK"] = "DK";
    Country["DM"] = "DM";
    Country["DO"] = "DO";
    Country["DZ"] = "DZ";
    Country["EC"] = "EC";
    Country["EE"] = "EE";
    Country["EG"] = "EG";
    Country["EH"] = "EH";
    Country["ER"] = "ER";
    Country["ES"] = "ES";
    Country["ET"] = "ET";
    Country["FI"] = "FI";
    Country["FJ"] = "FJ";
    Country["FK"] = "FK";
    Country["FM"] = "FM";
    Country["FO"] = "FO";
    Country["FR"] = "FR";
    Country["GA"] = "GA";
    Country["GB"] = "GB";
    Country["GD"] = "GD";
    Country["GE"] = "GE";
    Country["GF"] = "GF";
    Country["GG"] = "GG";
    Country["GH"] = "GH";
    Country["GI"] = "GI";
    Country["GL"] = "GL";
    Country["GM"] = "GM";
    Country["GN"] = "GN";
    Country["GP"] = "GP";
    Country["GQ"] = "GQ";
    Country["GR"] = "GR";
    Country["GS"] = "GS";
    Country["GT"] = "GT";
    Country["GU"] = "GU";
    Country["GW"] = "GW";
    Country["GY"] = "GY";
    Country["HK"] = "HK";
    Country["HN"] = "HN";
    Country["HR"] = "HR";
    Country["HT"] = "HT";
    Country["HU"] = "HU";
    Country["ID"] = "ID";
    Country["IE"] = "IE";
    Country["IL"] = "IL";
    Country["IM"] = "IM";
    Country["IN"] = "IN";
    Country["IO"] = "IO";
    Country["IQ"] = "IQ";
    Country["IR"] = "IR";
    Country["IS"] = "IS";
    Country["IT"] = "IT";
    Country["JE"] = "JE";
    Country["JM"] = "JM";
    Country["JO"] = "JO";
    Country["JP"] = "JP";
    Country["KE"] = "KE";
    Country["KG"] = "KG";
    Country["KH"] = "KH";
    Country["KI"] = "KI";
    Country["KM"] = "KM";
    Country["KN"] = "KN";
    Country["KP"] = "KP";
    Country["KR"] = "KR";
    Country["KW"] = "KW";
    Country["KY"] = "KY";
    Country["KZ"] = "KZ";
    Country["LA"] = "LA";
    Country["LB"] = "LB";
    Country["LC"] = "LC";
    Country["LI"] = "LI";
    Country["LK"] = "LK";
    Country["LR"] = "LR";
    Country["LS"] = "LS";
    Country["LT"] = "LT";
    Country["LU"] = "LU";
    Country["LV"] = "LV";
    Country["LY"] = "LY";
    Country["MA"] = "MA";
    Country["MC"] = "MC";
    Country["MD"] = "MD";
    Country["ME"] = "ME";
    Country["MF"] = "MF";
    Country["MG"] = "MG";
    Country["MH"] = "MH";
    Country["MK"] = "MK";
    Country["ML"] = "ML";
    Country["MM"] = "MM";
    Country["MN"] = "MN";
    Country["MO"] = "MO";
    Country["MP"] = "MP";
    Country["MQ"] = "MQ";
    Country["MR"] = "MR";
    Country["MS"] = "MS";
    Country["MT"] = "MT";
    Country["MU"] = "MU";
    Country["MV"] = "MV";
    Country["MW"] = "MW";
    Country["MX"] = "MX";
    Country["MY"] = "MY";
    Country["MZ"] = "MZ";
    Country["NA"] = "NA";
    Country["NC"] = "NC";
    Country["NE"] = "NE";
    Country["NF"] = "NF";
    Country["NG"] = "NG";
    Country["NI"] = "NI";
    Country["NL"] = "NL";
    Country["NO"] = "NO";
    Country["NP"] = "NP";
    Country["NR"] = "NR";
    Country["NU"] = "NU";
    Country["NZ"] = "NZ";
    Country["OM"] = "OM";
    Country["PA"] = "PA";
    Country["PE"] = "PE";
    Country["PF"] = "PF";
    Country["PG"] = "PG";
    Country["PH"] = "PH";
    Country["PK"] = "PK";
    Country["PL"] = "PL";
    Country["PM"] = "PM";
    Country["PN"] = "PN";
    Country["PR"] = "PR";
    Country["PS"] = "PS";
    Country["PT"] = "PT";
    Country["PW"] = "PW";
    Country["PY"] = "PY";
    Country["QA"] = "QA";
    Country["RE"] = "RE";
    Country["RO"] = "RO";
    Country["RS"] = "RS";
    Country["RU"] = "RU";
    Country["RW"] = "RW";
    Country["SA"] = "SA";
    Country["SB"] = "SB";
    Country["SC"] = "SC";
    Country["SD"] = "SD";
    Country["SE"] = "SE";
    Country["SG"] = "SG";
    Country["SH"] = "SH";
    Country["SI"] = "SI";
    Country["SJ"] = "SJ";
    Country["SK"] = "SK";
    Country["SL"] = "SL";
    Country["SM"] = "SM";
    Country["SN"] = "SN";
    Country["SO"] = "SO";
    Country["SR"] = "SR";
    Country["SS"] = "SS";
    Country["ST"] = "ST";
    Country["SV"] = "SV";
    Country["SX"] = "SX";
    Country["SY"] = "SY";
    Country["SZ"] = "SZ";
    Country["TC"] = "TC";
    Country["TD"] = "TD";
    Country["TG"] = "TG";
    Country["TH"] = "TH";
    Country["TJ"] = "TJ";
    Country["TK"] = "TK";
    Country["TL"] = "TL";
    Country["TM"] = "TM";
    Country["TN"] = "TN";
    Country["TO"] = "TO";
    Country["TR"] = "TR";
    Country["TT"] = "TT";
    Country["TV"] = "TV";
    Country["TW"] = "TW";
    Country["TZ"] = "TZ";
    Country["UA"] = "UA";
    Country["UG"] = "UG";
    Country["UM"] = "UM";
    Country["US"] = "US";
    Country["UY"] = "UY";
    Country["UZ"] = "UZ";
    Country["VA"] = "VA";
    Country["VC"] = "VC";
    Country["VE"] = "VE";
    Country["VG"] = "VG";
    Country["VI"] = "VI";
    Country["VN"] = "VN";
    Country["VU"] = "VU";
    Country["WF"] = "WF";
    Country["WS"] = "WS";
    Country["XK"] = "XK";
    Country["YE"] = "YE";
    Country["YT"] = "YT";
    Country["ZA"] = "ZA";
    Country["ZM"] = "ZM";
    Country["ZW"] = "ZW";
})(Country || (Country = {}));
/** One of 'AED', 'AFN', 'ALL', 'AMD', 'AOA', 'ARS', 'AUD', 'AWG', 'AZN', 'BAM', 'BBD', 'BDT', 'BGN', 'BHD', 'BIF', 'BMD', 'BND', 'BOB', 'BOV', 'BRL', 'BSD', 'BTC', 'BTN', 'BWP', 'BYN', 'BZD', 'CAD', 'CDF', 'CHE', 'CHF', 'CHW', 'CLF', 'CLP', 'CNY', 'COP', 'COU', 'CRC', 'CUP', 'CVE', 'CZK', 'DJF', 'DKK', 'DOP', 'DZD', 'EGP', 'ERN', 'ETB', 'ETH', 'EUR', 'FJD', 'FKP', 'GBP', 'GEL', 'GHS', 'GIP', 'GMD', 'GNF', 'GTQ', 'GYD', 'HKD', 'HNL', 'HTG', 'HUF', 'IDR', 'ILS', 'INR', 'IQD', 'IRR', 'ISK', 'JMD', 'JOD', 'JPY', 'KES', 'KGS', 'KHR', 'KMF', 'KPW', 'KRW', 'KWD', 'KYD', 'KZT', 'LAK', 'LBP', 'LKR', 'LRD', 'LSL', 'LYD', 'MAD', 'MDL', 'MGA', 'MKD', 'MMK', 'MNT', 'MOP', 'MRU', 'MUR', 'MVR', 'MWK', 'MXN', 'MXV', 'MYR', 'MZN', 'NAD', 'NGN', 'NIO', 'NOK', 'NPR', 'NZD', 'OMR', 'PAB', 'PEN', 'PGK', 'PHP', 'PKR', 'PLN', 'PYG', 'QAR', 'RON', 'RSD', 'RUB', 'RWF', 'SAR', 'SBD', 'SCR', 'SDG', 'SEK', 'SGD', 'SHP', 'SLE', 'SOS', 'SRD', 'SSP', 'STN', 'SVC', 'SYP', 'SZL', 'THB', 'TJS', 'TMT', 'TND', 'TOP', 'TRY', 'TTD', 'TWD', 'TZS', 'UAH', 'UGX', 'USD', 'USN', 'UYI', 'UYU', 'UYW', 'UZS', 'VED', 'VES', 'VND', 'VUV', 'WST', 'XAD', 'XAF', 'XAG', 'XAU', 'XBA', 'XBB', 'XBC', 'XBD', 'XCD', 'XCG', 'XDR', 'XOF', 'XPD', 'XPF', 'XPT', 'XSU', 'XTS', 'XUA', 'XXX', 'YER', 'ZAR', 'ZMW', 'ZWG' */
export var Currency;
(function (Currency) {
    Currency["AED"] = "AED";
    Currency["AFN"] = "AFN";
    Currency["ALL"] = "ALL";
    Currency["AMD"] = "AMD";
    Currency["AOA"] = "AOA";
    Currency["ARS"] = "ARS";
    Currency["AUD"] = "AUD";
    Currency["AWG"] = "AWG";
    Currency["AZN"] = "AZN";
    Currency["BAM"] = "BAM";
    Currency["BBD"] = "BBD";
    Currency["BDT"] = "BDT";
    Currency["BGN"] = "BGN";
    Currency["BHD"] = "BHD";
    Currency["BIF"] = "BIF";
    Currency["BMD"] = "BMD";
    Currency["BND"] = "BND";
    Currency["BOB"] = "BOB";
    Currency["BOV"] = "BOV";
    Currency["BRL"] = "BRL";
    Currency["BSD"] = "BSD";
    Currency["BTC"] = "BTC";
    Currency["BTN"] = "BTN";
    Currency["BWP"] = "BWP";
    Currency["BYN"] = "BYN";
    Currency["BZD"] = "BZD";
    Currency["CAD"] = "CAD";
    Currency["CDF"] = "CDF";
    Currency["CHE"] = "CHE";
    Currency["CHF"] = "CHF";
    Currency["CHW"] = "CHW";
    Currency["CLF"] = "CLF";
    Currency["CLP"] = "CLP";
    Currency["CNY"] = "CNY";
    Currency["COP"] = "COP";
    Currency["COU"] = "COU";
    Currency["CRC"] = "CRC";
    Currency["CUP"] = "CUP";
    Currency["CVE"] = "CVE";
    Currency["CZK"] = "CZK";
    Currency["DJF"] = "DJF";
    Currency["DKK"] = "DKK";
    Currency["DOP"] = "DOP";
    Currency["DZD"] = "DZD";
    Currency["EGP"] = "EGP";
    Currency["ERN"] = "ERN";
    Currency["ETB"] = "ETB";
    Currency["ETH"] = "ETH";
    Currency["EUR"] = "EUR";
    Currency["FJD"] = "FJD";
    Currency["FKP"] = "FKP";
    Currency["GBP"] = "GBP";
    Currency["GEL"] = "GEL";
    Currency["GHS"] = "GHS";
    Currency["GIP"] = "GIP";
    Currency["GMD"] = "GMD";
    Currency["GNF"] = "GNF";
    Currency["GTQ"] = "GTQ";
    Currency["GYD"] = "GYD";
    Currency["HKD"] = "HKD";
    Currency["HNL"] = "HNL";
    Currency["HTG"] = "HTG";
    Currency["HUF"] = "HUF";
    Currency["IDR"] = "IDR";
    Currency["ILS"] = "ILS";
    Currency["INR"] = "INR";
    Currency["IQD"] = "IQD";
    Currency["IRR"] = "IRR";
    Currency["ISK"] = "ISK";
    Currency["JMD"] = "JMD";
    Currency["JOD"] = "JOD";
    Currency["JPY"] = "JPY";
    Currency["KES"] = "KES";
    Currency["KGS"] = "KGS";
    Currency["KHR"] = "KHR";
    Currency["KMF"] = "KMF";
    Currency["KPW"] = "KPW";
    Currency["KRW"] = "KRW";
    Currency["KWD"] = "KWD";
    Currency["KYD"] = "KYD";
    Currency["KZT"] = "KZT";
    Currency["LAK"] = "LAK";
    Currency["LBP"] = "LBP";
    Currency["LKR"] = "LKR";
    Currency["LRD"] = "LRD";
    Currency["LSL"] = "LSL";
    Currency["LYD"] = "LYD";
    Currency["MAD"] = "MAD";
    Currency["MDL"] = "MDL";
    Currency["MGA"] = "MGA";
    Currency["MKD"] = "MKD";
    Currency["MMK"] = "MMK";
    Currency["MNT"] = "MNT";
    Currency["MOP"] = "MOP";
    Currency["MRU"] = "MRU";
    Currency["MUR"] = "MUR";
    Currency["MVR"] = "MVR";
    Currency["MWK"] = "MWK";
    Currency["MXN"] = "MXN";
    Currency["MXV"] = "MXV";
    Currency["MYR"] = "MYR";
    Currency["MZN"] = "MZN";
    Currency["NAD"] = "NAD";
    Currency["NGN"] = "NGN";
    Currency["NIO"] = "NIO";
    Currency["NOK"] = "NOK";
    Currency["NPR"] = "NPR";
    Currency["NZD"] = "NZD";
    Currency["OMR"] = "OMR";
    Currency["PAB"] = "PAB";
    Currency["PEN"] = "PEN";
    Currency["PGK"] = "PGK";
    Currency["PHP"] = "PHP";
    Currency["PKR"] = "PKR";
    Currency["PLN"] = "PLN";
    Currency["PYG"] = "PYG";
    Currency["QAR"] = "QAR";
    Currency["RON"] = "RON";
    Currency["RSD"] = "RSD";
    Currency["RUB"] = "RUB";
    Currency["RWF"] = "RWF";
    Currency["SAR"] = "SAR";
    Currency["SBD"] = "SBD";
    Currency["SCR"] = "SCR";
    Currency["SDG"] = "SDG";
    Currency["SEK"] = "SEK";
    Currency["SGD"] = "SGD";
    Currency["SHP"] = "SHP";
    Currency["SLE"] = "SLE";
    Currency["SOS"] = "SOS";
    Currency["SRD"] = "SRD";
    Currency["SSP"] = "SSP";
    Currency["STN"] = "STN";
    Currency["SVC"] = "SVC";
    Currency["SYP"] = "SYP";
    Currency["SZL"] = "SZL";
    Currency["THB"] = "THB";
    Currency["TJS"] = "TJS";
    Currency["TMT"] = "TMT";
    Currency["TND"] = "TND";
    Currency["TOP"] = "TOP";
    Currency["TRY"] = "TRY";
    Currency["TTD"] = "TTD";
    Currency["TWD"] = "TWD";
    Currency["TZS"] = "TZS";
    Currency["UAH"] = "UAH";
    Currency["UGX"] = "UGX";
    Currency["USD"] = "USD";
    Currency["USN"] = "USN";
    Currency["UYI"] = "UYI";
    Currency["UYU"] = "UYU";
    Currency["UYW"] = "UYW";
    Currency["UZS"] = "UZS";
    Currency["VED"] = "VED";
    Currency["VES"] = "VES";
    Currency["VND"] = "VND";
    Currency["VUV"] = "VUV";
    Currency["WST"] = "WST";
    Currency["XAD"] = "XAD";
    Currency["XAF"] = "XAF";
    Currency["XAG"] = "XAG";
    Currency["XAU"] = "XAU";
    Currency["XBA"] = "XBA";
    Currency["XBB"] = "XBB";
    Currency["XBC"] = "XBC";
    Currency["XBD"] = "XBD";
    Currency["XCD"] = "XCD";
    Currency["XCG"] = "XCG";
    Currency["XDR"] = "XDR";
    Currency["XOF"] = "XOF";
    Currency["XPD"] = "XPD";
    Currency["XPF"] = "XPF";
    Currency["XPT"] = "XPT";
    Currency["XSU"] = "XSU";
    Currency["XTS"] = "XTS";
    Currency["XUA"] = "XUA";
    Currency["XXX"] = "XXX";
    Currency["YER"] = "YER";
    Currency["ZAR"] = "ZAR";
    Currency["ZMW"] = "ZMW";
    Currency["ZWG"] = "ZWG";
})(Currency || (Currency = {}));
/** One of 'dmy_dashes', 'dmy_slashes', 'mdy_dashes', 'mdy_slashes', 'ymd_dashes', 'ymd_slashes' */
export var DateFormat;
(function (DateFormat) {
    DateFormat["Dmy_dashes"] = "dmy_dashes";
    DateFormat["Dmy_slashes"] = "dmy_slashes";
    DateFormat["Mdy_dashes"] = "mdy_dashes";
    DateFormat["Mdy_slashes"] = "mdy_slashes";
    DateFormat["Ymd_dashes"] = "ymd_dashes";
    DateFormat["Ymd_slashes"] = "ymd_slashes";
})(DateFormat || (DateFormat = {}));
/** One of 'apple', 'google' */
export var DigitalWalletType;
(function (DigitalWalletType) {
    DigitalWalletType["Apple"] = "apple";
    DigitalWalletType["Google"] = "google";
})(DigitalWalletType || (DigitalWalletType = {}));
export var IsoDayOfWeek;
(function (IsoDayOfWeek) {
    IsoDayOfWeek[IsoDayOfWeek["_0"] = 0] = "_0";
    IsoDayOfWeek[IsoDayOfWeek["_1"] = 1] = "_1";
    IsoDayOfWeek[IsoDayOfWeek["_2"] = 2] = "_2";
    IsoDayOfWeek[IsoDayOfWeek["_3"] = 3] = "_3";
    IsoDayOfWeek[IsoDayOfWeek["_4"] = 4] = "_4";
    IsoDayOfWeek[IsoDayOfWeek["_5"] = 5] = "_5";
    IsoDayOfWeek[IsoDayOfWeek["_6"] = 6] = "_6";
    IsoDayOfWeek[IsoDayOfWeek["_7"] = 7] = "_7";
})(IsoDayOfWeek || (IsoDayOfWeek = {}));
/** One of 'eu1', 'eu2', 'international' */
export var NumberFormat;
(function (NumberFormat) {
    NumberFormat["Eu1"] = "eu1";
    NumberFormat["Eu2"] = "eu2";
    NumberFormat["International"] = "international";
})(NumberFormat || (NumberFormat = {}));
/** One of 'ecommerce', 'moto' */
export var PaymentChannel;
(function (PaymentChannel) {
    PaymentChannel["Ecommerce"] = "ecommerce";
    PaymentChannel["Moto"] = "moto";
})(PaymentChannel || (PaymentChannel = {}));
/** One of 'en', 'es', 'fr', 'xx' */
export var SystemLanguage;
(function (SystemLanguage) {
    SystemLanguage["En"] = "en";
    SystemLanguage["Es"] = "es";
    SystemLanguage["Fr"] = "fr";
    SystemLanguage["Xx"] = "xx";
})(SystemLanguage || (SystemLanguage = {}));
/** One of '12', '24' */
export var TimeFormat;
(function (TimeFormat) {
    TimeFormat["_12"] = "12";
    TimeFormat["_24"] = "24";
})(TimeFormat || (TimeFormat = {}));
var ApiException = /** @class */ (function (_super) {
    __extends(ApiException, _super);
    function ApiException(message, status, response, headers, result) {
        var _this = _super.call(this) || this;
        _this.isApiException = true;
        _this.message = message;
        _this.status = status;
        _this.response = response;
        _this.headers = headers;
        _this.result = result;
        return _this;
    }
    ApiException.isApiException = function (obj) {
        return obj.isApiException === true;
    };
    return ApiException;
}(Error));
export { ApiException };
function throwException(message, status, response, headers, result) {
    if (result !== null && result !== undefined)
        throw result;
    else
        throw new ApiException(message, status, response, headers, null);
}
