// @n3oltd/karakoram.platforms.sdk.connect/esm/index.js
//----------------------
// <auto-generated>
//     Generated using the NSwag toolchain v14.6.0.0 (NJsonSchema v11.5.0.0 (Newtonsoft.Json v13.0.0.0)) (http://NSwag.org)
// </auto-generated>
//----------------------
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
/* eslint-disable */
// ReSharper disable InconsistentNaming
var ConnectApiBase = /** @class */ (function () {
    function ConnectApiBase(apiHeaders) {
        var _this = this;
        this.transformOptions = function (options) {
            var _a, _b;
            var headers = __assign({}, options.headers);
            var subscriptionId = _this.apiHeaders.getSubscriptionId();
            if (subscriptionId) {
                headers['N3O-Subscription-Id'] = subscriptionId;
            }
            var bearerToken = (_b = (_a = _this.apiHeaders).getBearerToken) === null || _b === void 0 ? void 0 : _b.call(_a);
            if (bearerToken) {
                headers['Authorization'] = bearerToken;
            }
            options.headers = headers;
            return Promise.resolve(options);
        };
        this.apiHeaders = apiHeaders;
    }
    return ConnectApiBase;
}());
export { ConnectApiBase };
var ConnectClient = /** @class */ (function (_super) {
    __extends(ConnectClient, _super);
    function ConnectClient(configuration, baseUrl, http) {
        var _this = _super.call(this, configuration) || this;
        _this.jsonParseReviver = undefined;
        _this.http = http ? http : window;
        _this.baseUrl = baseUrl !== null && baseUrl !== void 0 ? baseUrl : "/";
        return _this;
    }
    /**
     * @param body (optional)
     * @return OK
     */
    ConnectClient.prototype.search = function (body) {
        var _this = this;
        var url_ = this.baseUrl + "/v1.0/search";
        url_ = url_.replace(/[?&]$/, "");
        var content_ = JSON.stringify(body);
        var options_ = {
            body: content_,
            method: "POST",
            headers: {
                "Content-Type": "application/json",
                "Accept": "application/json"
            }
        };
        return this.transformOptions(options_).then(function (transformedOptions_) {
            return _this.http.fetch(url_, transformedOptions_);
        }).then(function (_response) {
            return _this.processSearch(_response);
        });
    };
    ConnectClient.prototype.processSearch = function (response) {
        var _this = this;
        var status = response.status;
        var _headers = {};
        if (response.headers && response.headers.forEach) {
            response.headers.forEach(function (v, k) { return _headers[k] = v; });
        }
        ;
        if (status === 200) {
            return response.text().then(function (_responseText) {
                var result200 = null;
                result200 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return result200;
            });
        }
        else if (status === 204) {
            return response.text().then(function (_responseText) {
                return throwException("No Content", status, _responseText, _headers);
            });
        }
        else if (status === 400) {
            return response.text().then(function (_responseText) {
                var result400 = null;
                result400 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return throwException("Bad Request", status, _responseText, _headers, result400);
            });
        }
        else if (status === 401) {
            return response.text().then(function (_responseText) {
                var result401 = null;
                result401 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return throwException("Unauthorized", status, _responseText, _headers, result401);
            });
        }
        else if (status === 404) {
            return response.text().then(function (_responseText) {
                var result404 = null;
                result404 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return throwException("Not Found", status, _responseText, _headers, result404);
            });
        }
        else if (status === 412) {
            return response.text().then(function (_responseText) {
                var result412 = null;
                result412 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return throwException("Precondition Failed", status, _responseText, _headers, result412);
            });
        }
        else if (status === 500) {
            return response.text().then(function (_responseText) {
                return throwException("Internal Server Error", status, _responseText, _headers);
            });
        }
        else if (status !== 200 && status !== 204) {
            return response.text().then(function (_responseText) {
                return throwException("An unexpected server error occurred.", status, _responseText, _headers);
            });
        }
        return Promise.resolve(null);
    };
    return ConnectClient;
}(ConnectApiBase));
export { ConnectClient };
var ApiException = /** @class */ (function (_super) {
    __extends(ApiException, _super);
    function ApiException(message, status, response, headers, result) {
        var _this = _super.call(this) || this;
        _this.isApiException = true;
        _this.message = message;
        _this.status = status;
        _this.response = response;
        _this.headers = headers;
        _this.result = result;
        return _this;
    }
    ApiException.isApiException = function (obj) {
        return obj.isApiException === true;
    };
    return ApiException;
}(Error));
export { ApiException };
function throwException(message, status, response, headers, result) {
    if (result !== null && result !== undefined)
        throw result;
    else
        throw new ApiException(message, status, response, headers, null);
}
