// @n3oltd/karakoram.checkout.sdk.payit/esm/index.js
//----------------------
// <auto-generated>
//     Generated using the NSwag toolchain v14.6.0.0 (NJsonSchema v11.5.0.0 (Newtonsoft.Json v13.0.0.0)) (http://NSwag.org)
// </auto-generated>
//----------------------
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
/* eslint-disable */
// ReSharper disable InconsistentNaming
var ConnectApiBase = /** @class */ (function () {
    function ConnectApiBase(apiHeaders) {
        var _this = this;
        this.transformOptions = function (options) {
            var _a, _b;
            var headers = __assign({}, options.headers);
            var subscriptionId = _this.apiHeaders.getSubscriptionId();
            if (subscriptionId) {
                headers['N3O-Subscription-Id'] = subscriptionId;
            }
            var bearerToken = (_b = (_a = _this.apiHeaders).getBearerToken) === null || _b === void 0 ? void 0 : _b.call(_a);
            if (bearerToken) {
                headers['Authorization'] = bearerToken;
            }
            options.headers = headers;
            return Promise.resolve(options);
        };
        this.apiHeaders = apiHeaders;
    }
    return ConnectApiBase;
}());
export { ConnectApiBase };
var PayitClient = /** @class */ (function (_super) {
    __extends(PayitClient, _super);
    function PayitClient(configuration, baseUrl, http) {
        var _this = _super.call(this, configuration) || this;
        _this.jsonParseReviver = undefined;
        _this.http = http ? http : window;
        _this.baseUrl = baseUrl !== null && baseUrl !== void 0 ? baseUrl : "/";
        return _this;
    }
    /**
     * @return OK
     */
    PayitClient.prototype.getEligibleBanks = function (methodId) {
        var _this = this;
        var url_ = this.baseUrl + "/payit/v1.0/methods/{methodId}/eligiblebanks";
        if (methodId === undefined || methodId === null)
            throw new globalThis.Error("The parameter 'methodId' must be defined.");
        url_ = url_.replace("{methodId}", encodeURIComponent("" + methodId));
        url_ = url_.replace(/[?&]$/, "");
        var options_ = {
            method: "POST",
            headers: {
                "Accept": "application/json"
            }
        };
        return this.transformOptions(options_).then(function (transformedOptions_) {
            return _this.http.fetch(url_, transformedOptions_);
        }).then(function (_response) {
            return _this.processGetEligibleBanks(_response);
        });
    };
    PayitClient.prototype.processGetEligibleBanks = function (response) {
        var _this = this;
        var status = response.status;
        var _headers = {};
        if (response.headers && response.headers.forEach) {
            response.headers.forEach(function (v, k) { return _headers[k] = v; });
        }
        ;
        if (status === 200) {
            return response.text().then(function (_responseText) {
                var result200 = null;
                result200 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return result200;
            });
        }
        else if (status === 204) {
            return response.text().then(function (_responseText) {
                return throwException("No Content", status, _responseText, _headers);
            });
        }
        else if (status === 400) {
            return response.text().then(function (_responseText) {
                var result400 = null;
                result400 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return throwException("Bad Request", status, _responseText, _headers, result400);
            });
        }
        else if (status === 401) {
            return response.text().then(function (_responseText) {
                var result401 = null;
                result401 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return throwException("Unauthorized", status, _responseText, _headers, result401);
            });
        }
        else if (status === 404) {
            return response.text().then(function (_responseText) {
                var result404 = null;
                result404 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return throwException("Not Found", status, _responseText, _headers, result404);
            });
        }
        else if (status === 412) {
            return response.text().then(function (_responseText) {
                var result412 = null;
                result412 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return throwException("Precondition Failed", status, _responseText, _headers, result412);
            });
        }
        else if (status === 500) {
            return response.text().then(function (_responseText) {
                return throwException("Internal Server Error", status, _responseText, _headers);
            });
        }
        else if (status !== 200 && status !== 204) {
            return response.text().then(function (_responseText) {
                return throwException("An unexpected server error occurred.", status, _responseText, _headers);
            });
        }
        return Promise.resolve(null);
    };
    /**
     * @return OK
     */
    PayitClient.prototype.isProcessing = function (methodId, flowId, scopeId, paymentId) {
        var _this = this;
        var url_ = this.baseUrl + "/payit/v1.0/payments/{methodId}/{flowId}/{scopeId}/{paymentId}/isprocessing";
        if (methodId === undefined || methodId === null)
            throw new globalThis.Error("The parameter 'methodId' must be defined.");
        url_ = url_.replace("{methodId}", encodeURIComponent("" + methodId));
        if (flowId === undefined || flowId === null)
            throw new globalThis.Error("The parameter 'flowId' must be defined.");
        url_ = url_.replace("{flowId}", encodeURIComponent("" + flowId));
        if (scopeId === undefined || scopeId === null)
            throw new globalThis.Error("The parameter 'scopeId' must be defined.");
        url_ = url_.replace("{scopeId}", encodeURIComponent("" + scopeId));
        if (paymentId === undefined || paymentId === null)
            throw new globalThis.Error("The parameter 'paymentId' must be defined.");
        url_ = url_.replace("{paymentId}", encodeURIComponent("" + paymentId));
        url_ = url_.replace(/[?&]$/, "");
        var options_ = {
            method: "GET",
            headers: {
                "Accept": "application/json"
            }
        };
        return this.transformOptions(options_).then(function (transformedOptions_) {
            return _this.http.fetch(url_, transformedOptions_);
        }).then(function (_response) {
            return _this.processIsProcessing(_response);
        });
    };
    PayitClient.prototype.processIsProcessing = function (response) {
        var _this = this;
        var status = response.status;
        var _headers = {};
        if (response.headers && response.headers.forEach) {
            response.headers.forEach(function (v, k) { return _headers[k] = v; });
        }
        ;
        if (status === 200) {
            return response.text().then(function (_responseText) {
                var result200 = null;
                result200 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return result200;
            });
        }
        else if (status === 204) {
            return response.text().then(function (_responseText) {
                return throwException("No Content", status, _responseText, _headers);
            });
        }
        else if (status === 400) {
            return response.text().then(function (_responseText) {
                var result400 = null;
                result400 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return throwException("Bad Request", status, _responseText, _headers, result400);
            });
        }
        else if (status === 401) {
            return response.text().then(function (_responseText) {
                var result401 = null;
                result401 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return throwException("Unauthorized", status, _responseText, _headers, result401);
            });
        }
        else if (status === 404) {
            return response.text().then(function (_responseText) {
                var result404 = null;
                result404 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return throwException("Not Found", status, _responseText, _headers, result404);
            });
        }
        else if (status === 412) {
            return response.text().then(function (_responseText) {
                var result412 = null;
                result412 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return throwException("Precondition Failed", status, _responseText, _headers, result412);
            });
        }
        else if (status === 500) {
            return response.text().then(function (_responseText) {
                return throwException("Internal Server Error", status, _responseText, _headers);
            });
        }
        else if (status !== 200 && status !== 204) {
            return response.text().then(function (_responseText) {
                return throwException("An unexpected server error occurred.", status, _responseText, _headers);
            });
        }
        return Promise.resolve(null);
    };
    /**
     * @param body (optional)
     * @return OK
     */
    PayitClient.prototype.start = function (methodId, flowId, scopeId, body) {
        var _this = this;
        var url_ = this.baseUrl + "/payit/v1.0/payments/{methodId}/{flowId}/{scopeId}/start";
        if (methodId === undefined || methodId === null)
            throw new globalThis.Error("The parameter 'methodId' must be defined.");
        url_ = url_.replace("{methodId}", encodeURIComponent("" + methodId));
        if (flowId === undefined || flowId === null)
            throw new globalThis.Error("The parameter 'flowId' must be defined.");
        url_ = url_.replace("{flowId}", encodeURIComponent("" + flowId));
        if (scopeId === undefined || scopeId === null)
            throw new globalThis.Error("The parameter 'scopeId' must be defined.");
        url_ = url_.replace("{scopeId}", encodeURIComponent("" + scopeId));
        url_ = url_.replace(/[?&]$/, "");
        var content_ = JSON.stringify(body);
        var options_ = {
            body: content_,
            method: "POST",
            headers: {
                "Content-Type": "application/json",
                "Accept": "application/json"
            }
        };
        return this.transformOptions(options_).then(function (transformedOptions_) {
            return _this.http.fetch(url_, transformedOptions_);
        }).then(function (_response) {
            return _this.processStart(_response);
        });
    };
    PayitClient.prototype.processStart = function (response) {
        var _this = this;
        var status = response.status;
        var _headers = {};
        if (response.headers && response.headers.forEach) {
            response.headers.forEach(function (v, k) { return _headers[k] = v; });
        }
        ;
        if (status === 200) {
            return response.text().then(function (_responseText) {
                var result200 = null;
                result200 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return result200;
            });
        }
        else if (status === 204) {
            return response.text().then(function (_responseText) {
                return throwException("No Content", status, _responseText, _headers);
            });
        }
        else if (status === 400) {
            return response.text().then(function (_responseText) {
                var result400 = null;
                result400 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return throwException("Bad Request", status, _responseText, _headers, result400);
            });
        }
        else if (status === 401) {
            return response.text().then(function (_responseText) {
                var result401 = null;
                result401 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return throwException("Unauthorized", status, _responseText, _headers, result401);
            });
        }
        else if (status === 404) {
            return response.text().then(function (_responseText) {
                var result404 = null;
                result404 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return throwException("Not Found", status, _responseText, _headers, result404);
            });
        }
        else if (status === 412) {
            return response.text().then(function (_responseText) {
                var result412 = null;
                result412 = _responseText === "" ? null : JSON.parse(_responseText, _this.jsonParseReviver);
                return throwException("Precondition Failed", status, _responseText, _headers, result412);
            });
        }
        else if (status === 500) {
            return response.text().then(function (_responseText) {
                return throwException("Internal Server Error", status, _responseText, _headers);
            });
        }
        else if (status !== 200 && status !== 204) {
            return response.text().then(function (_responseText) {
                return throwException("An unexpected server error occurred.", status, _responseText, _headers);
            });
        }
        return Promise.resolve(null);
    };
    return PayitClient;
}(ConnectApiBase));
export { PayitClient };
/** One of 'AED', 'AFN', 'ALL', 'AMD', 'AOA', 'ARS', 'AUD', 'AWG', 'AZN', 'BAM', 'BBD', 'BDT', 'BGN', 'BHD', 'BIF', 'BMD', 'BND', 'BOB', 'BOV', 'BRL', 'BSD', 'BTC', 'BTN', 'BWP', 'BYN', 'BZD', 'CAD', 'CDF', 'CHE', 'CHF', 'CHW', 'CLF', 'CLP', 'CNY', 'COP', 'COU', 'CRC', 'CUP', 'CVE', 'CZK', 'DJF', 'DKK', 'DOP', 'DZD', 'EGP', 'ERN', 'ETB', 'ETH', 'EUR', 'FJD', 'FKP', 'GBP', 'GEL', 'GHS', 'GIP', 'GMD', 'GNF', 'GTQ', 'GYD', 'HKD', 'HNL', 'HTG', 'HUF', 'IDR', 'ILS', 'INR', 'IQD', 'IRR', 'ISK', 'JMD', 'JOD', 'JPY', 'KES', 'KGS', 'KHR', 'KMF', 'KPW', 'KRW', 'KWD', 'KYD', 'KZT', 'LAK', 'LBP', 'LKR', 'LRD', 'LSL', 'LYD', 'MAD', 'MDL', 'MGA', 'MKD', 'MMK', 'MNT', 'MOP', 'MRU', 'MUR', 'MVR', 'MWK', 'MXN', 'MXV', 'MYR', 'MZN', 'NAD', 'NGN', 'NIO', 'NOK', 'NPR', 'NZD', 'OMR', 'PAB', 'PEN', 'PGK', 'PHP', 'PKR', 'PLN', 'PYG', 'QAR', 'RON', 'RSD', 'RUB', 'RWF', 'SAR', 'SBD', 'SCR', 'SDG', 'SEK', 'SGD', 'SHP', 'SLE', 'SOS', 'SRD', 'SSP', 'STN', 'SVC', 'SYP', 'SZL', 'THB', 'TJS', 'TMT', 'TND', 'TOP', 'TRY', 'TTD', 'TWD', 'TZS', 'UAH', 'UGX', 'USD', 'USN', 'UYI', 'UYU', 'UYW', 'UZS', 'VED', 'VES', 'VND', 'VUV', 'WST', 'XAD', 'XAF', 'XAG', 'XAU', 'XBA', 'XBB', 'XBC', 'XBD', 'XCD', 'XCG', 'XDR', 'XOF', 'XPD', 'XPF', 'XPT', 'XSU', 'XTS', 'XUA', 'XXX', 'YER', 'ZAR', 'ZMW', 'ZWG' */
export var Currency;
(function (Currency) {
    Currency["AED"] = "AED";
    Currency["AFN"] = "AFN";
    Currency["ALL"] = "ALL";
    Currency["AMD"] = "AMD";
    Currency["AOA"] = "AOA";
    Currency["ARS"] = "ARS";
    Currency["AUD"] = "AUD";
    Currency["AWG"] = "AWG";
    Currency["AZN"] = "AZN";
    Currency["BAM"] = "BAM";
    Currency["BBD"] = "BBD";
    Currency["BDT"] = "BDT";
    Currency["BGN"] = "BGN";
    Currency["BHD"] = "BHD";
    Currency["BIF"] = "BIF";
    Currency["BMD"] = "BMD";
    Currency["BND"] = "BND";
    Currency["BOB"] = "BOB";
    Currency["BOV"] = "BOV";
    Currency["BRL"] = "BRL";
    Currency["BSD"] = "BSD";
    Currency["BTC"] = "BTC";
    Currency["BTN"] = "BTN";
    Currency["BWP"] = "BWP";
    Currency["BYN"] = "BYN";
    Currency["BZD"] = "BZD";
    Currency["CAD"] = "CAD";
    Currency["CDF"] = "CDF";
    Currency["CHE"] = "CHE";
    Currency["CHF"] = "CHF";
    Currency["CHW"] = "CHW";
    Currency["CLF"] = "CLF";
    Currency["CLP"] = "CLP";
    Currency["CNY"] = "CNY";
    Currency["COP"] = "COP";
    Currency["COU"] = "COU";
    Currency["CRC"] = "CRC";
    Currency["CUP"] = "CUP";
    Currency["CVE"] = "CVE";
    Currency["CZK"] = "CZK";
    Currency["DJF"] = "DJF";
    Currency["DKK"] = "DKK";
    Currency["DOP"] = "DOP";
    Currency["DZD"] = "DZD";
    Currency["EGP"] = "EGP";
    Currency["ERN"] = "ERN";
    Currency["ETB"] = "ETB";
    Currency["ETH"] = "ETH";
    Currency["EUR"] = "EUR";
    Currency["FJD"] = "FJD";
    Currency["FKP"] = "FKP";
    Currency["GBP"] = "GBP";
    Currency["GEL"] = "GEL";
    Currency["GHS"] = "GHS";
    Currency["GIP"] = "GIP";
    Currency["GMD"] = "GMD";
    Currency["GNF"] = "GNF";
    Currency["GTQ"] = "GTQ";
    Currency["GYD"] = "GYD";
    Currency["HKD"] = "HKD";
    Currency["HNL"] = "HNL";
    Currency["HTG"] = "HTG";
    Currency["HUF"] = "HUF";
    Currency["IDR"] = "IDR";
    Currency["ILS"] = "ILS";
    Currency["INR"] = "INR";
    Currency["IQD"] = "IQD";
    Currency["IRR"] = "IRR";
    Currency["ISK"] = "ISK";
    Currency["JMD"] = "JMD";
    Currency["JOD"] = "JOD";
    Currency["JPY"] = "JPY";
    Currency["KES"] = "KES";
    Currency["KGS"] = "KGS";
    Currency["KHR"] = "KHR";
    Currency["KMF"] = "KMF";
    Currency["KPW"] = "KPW";
    Currency["KRW"] = "KRW";
    Currency["KWD"] = "KWD";
    Currency["KYD"] = "KYD";
    Currency["KZT"] = "KZT";
    Currency["LAK"] = "LAK";
    Currency["LBP"] = "LBP";
    Currency["LKR"] = "LKR";
    Currency["LRD"] = "LRD";
    Currency["LSL"] = "LSL";
    Currency["LYD"] = "LYD";
    Currency["MAD"] = "MAD";
    Currency["MDL"] = "MDL";
    Currency["MGA"] = "MGA";
    Currency["MKD"] = "MKD";
    Currency["MMK"] = "MMK";
    Currency["MNT"] = "MNT";
    Currency["MOP"] = "MOP";
    Currency["MRU"] = "MRU";
    Currency["MUR"] = "MUR";
    Currency["MVR"] = "MVR";
    Currency["MWK"] = "MWK";
    Currency["MXN"] = "MXN";
    Currency["MXV"] = "MXV";
    Currency["MYR"] = "MYR";
    Currency["MZN"] = "MZN";
    Currency["NAD"] = "NAD";
    Currency["NGN"] = "NGN";
    Currency["NIO"] = "NIO";
    Currency["NOK"] = "NOK";
    Currency["NPR"] = "NPR";
    Currency["NZD"] = "NZD";
    Currency["OMR"] = "OMR";
    Currency["PAB"] = "PAB";
    Currency["PEN"] = "PEN";
    Currency["PGK"] = "PGK";
    Currency["PHP"] = "PHP";
    Currency["PKR"] = "PKR";
    Currency["PLN"] = "PLN";
    Currency["PYG"] = "PYG";
    Currency["QAR"] = "QAR";
    Currency["RON"] = "RON";
    Currency["RSD"] = "RSD";
    Currency["RUB"] = "RUB";
    Currency["RWF"] = "RWF";
    Currency["SAR"] = "SAR";
    Currency["SBD"] = "SBD";
    Currency["SCR"] = "SCR";
    Currency["SDG"] = "SDG";
    Currency["SEK"] = "SEK";
    Currency["SGD"] = "SGD";
    Currency["SHP"] = "SHP";
    Currency["SLE"] = "SLE";
    Currency["SOS"] = "SOS";
    Currency["SRD"] = "SRD";
    Currency["SSP"] = "SSP";
    Currency["STN"] = "STN";
    Currency["SVC"] = "SVC";
    Currency["SYP"] = "SYP";
    Currency["SZL"] = "SZL";
    Currency["THB"] = "THB";
    Currency["TJS"] = "TJS";
    Currency["TMT"] = "TMT";
    Currency["TND"] = "TND";
    Currency["TOP"] = "TOP";
    Currency["TRY"] = "TRY";
    Currency["TTD"] = "TTD";
    Currency["TWD"] = "TWD";
    Currency["TZS"] = "TZS";
    Currency["UAH"] = "UAH";
    Currency["UGX"] = "UGX";
    Currency["USD"] = "USD";
    Currency["USN"] = "USN";
    Currency["UYI"] = "UYI";
    Currency["UYU"] = "UYU";
    Currency["UYW"] = "UYW";
    Currency["UZS"] = "UZS";
    Currency["VED"] = "VED";
    Currency["VES"] = "VES";
    Currency["VND"] = "VND";
    Currency["VUV"] = "VUV";
    Currency["WST"] = "WST";
    Currency["XAD"] = "XAD";
    Currency["XAF"] = "XAF";
    Currency["XAG"] = "XAG";
    Currency["XAU"] = "XAU";
    Currency["XBA"] = "XBA";
    Currency["XBB"] = "XBB";
    Currency["XBC"] = "XBC";
    Currency["XBD"] = "XBD";
    Currency["XCD"] = "XCD";
    Currency["XCG"] = "XCG";
    Currency["XDR"] = "XDR";
    Currency["XOF"] = "XOF";
    Currency["XPD"] = "XPD";
    Currency["XPF"] = "XPF";
    Currency["XPT"] = "XPT";
    Currency["XSU"] = "XSU";
    Currency["XTS"] = "XTS";
    Currency["XUA"] = "XUA";
    Currency["XXX"] = "XXX";
    Currency["YER"] = "YER";
    Currency["ZAR"] = "ZAR";
    Currency["ZMW"] = "ZMW";
    Currency["ZWG"] = "ZWG";
})(Currency || (Currency = {}));
export var IsoDayOfWeek;
(function (IsoDayOfWeek) {
    IsoDayOfWeek[IsoDayOfWeek["_0"] = 0] = "_0";
    IsoDayOfWeek[IsoDayOfWeek["_1"] = 1] = "_1";
    IsoDayOfWeek[IsoDayOfWeek["_2"] = 2] = "_2";
    IsoDayOfWeek[IsoDayOfWeek["_3"] = 3] = "_3";
    IsoDayOfWeek[IsoDayOfWeek["_4"] = 4] = "_4";
    IsoDayOfWeek[IsoDayOfWeek["_5"] = 5] = "_5";
    IsoDayOfWeek[IsoDayOfWeek["_6"] = 6] = "_6";
    IsoDayOfWeek[IsoDayOfWeek["_7"] = 7] = "_7";
})(IsoDayOfWeek || (IsoDayOfWeek = {}));
var ApiException = /** @class */ (function (_super) {
    __extends(ApiException, _super);
    function ApiException(message, status, response, headers, result) {
        var _this = _super.call(this) || this;
        _this.isApiException = true;
        _this.message = message;
        _this.status = status;
        _this.response = response;
        _this.headers = headers;
        _this.result = result;
        return _this;
    }
    ApiException.isApiException = function (obj) {
        return obj.isApiException === true;
    };
    return ApiException;
}(Error));
export { ApiException };
function throwException(message, status, response, headers, result) {
    if (result !== null && result !== undefined)
        throw result;
    else
        throw new ApiException(message, status, response, headers, null);
}
