// @n3oltd/karakoram.connect.sdk.types/esm/index.js
//----------------------
// <auto-generated>
//     Generated using the NSwag toolchain v14.6.0.0 (NJsonSchema v11.5.0.0 (Newtonsoft.Json v13.0.0.0)) (http://NSwag.org)
// </auto-generated>
//----------------------
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
/* eslint-disable */
// ReSharper disable InconsistentNaming
var ConnectApiBase = /** @class */ (function () {
    function ConnectApiBase(apiHeaders) {
        var _this = this;
        this.transformOptions = function (options) {
            var _a, _b;
            var headers = __assign({}, options.headers);
            var subscriptionId = _this.apiHeaders.getSubscriptionId();
            if (subscriptionId) {
                headers['N3O-Subscription-Id'] = subscriptionId;
            }
            var bearerToken = (_b = (_a = _this.apiHeaders).getBearerToken) === null || _b === void 0 ? void 0 : _b.call(_a);
            if (bearerToken) {
                headers['Authorization'] = bearerToken;
            }
            options.headers = headers;
            return Promise.resolve(options);
        };
        this.apiHeaders = apiHeaders;
    }
    return ConnectApiBase;
}());
export { ConnectApiBase };
/** One of 'layout1', 'layout2' */
export var AddressLayout;
(function (AddressLayout) {
    AddressLayout["Layout1"] = "layout1";
    AddressLayout["Layout2"] = "layout2";
})(AddressLayout || (AddressLayout = {}));
/** One of 'email', 'post', 'sms', 'telephone', 'whatsApp' */
export var CommunicationChannel;
(function (CommunicationChannel) {
    CommunicationChannel["Email"] = "email";
    CommunicationChannel["Post"] = "post";
    CommunicationChannel["Sms"] = "sms";
    CommunicationChannel["Telephone"] = "telephone";
    CommunicationChannel["WhatsApp"] = "whatsApp";
})(CommunicationChannel || (CommunicationChannel = {}));
/** One of 'accountEntry', 'currencies', 'donationItems', 'environment', 'feedbackSchemes', 'fundStructure', 'localization', 'lookups', 'nisab', 'organizationInfo', 'payments', 'sponsorshipSchemes', 'tracking', 'typesense' */
export var ConnectSubscriptionFile;
(function (ConnectSubscriptionFile) {
    ConnectSubscriptionFile["AccountEntry"] = "accountEntry";
    ConnectSubscriptionFile["Currencies"] = "currencies";
    ConnectSubscriptionFile["DonationItems"] = "donationItems";
    ConnectSubscriptionFile["Environment"] = "environment";
    ConnectSubscriptionFile["FeedbackSchemes"] = "feedbackSchemes";
    ConnectSubscriptionFile["FundStructure"] = "fundStructure";
    ConnectSubscriptionFile["Localization"] = "localization";
    ConnectSubscriptionFile["Lookups"] = "lookups";
    ConnectSubscriptionFile["Nisab"] = "nisab";
    ConnectSubscriptionFile["OrganizationInfo"] = "organizationInfo";
    ConnectSubscriptionFile["Payments"] = "payments";
    ConnectSubscriptionFile["SponsorshipSchemes"] = "sponsorshipSchemes";
    ConnectSubscriptionFile["Tracking"] = "tracking";
    ConnectSubscriptionFile["Typesense"] = "typesense";
})(ConnectSubscriptionFile || (ConnectSubscriptionFile = {}));
/** One of 'AD', 'AE', 'AF', 'AG', 'AI', 'AL', 'AM', 'AN', 'AO', 'AR', 'AS', 'AT', 'AU', 'AW', 'AX', 'AZ', 'BA', 'BB', 'BD', 'BE', 'BF', 'BG', 'BH', 'BI', 'BJ', 'BL', 'BM', 'BN', 'BO', 'BQ', 'BR', 'BS', 'BT', 'BW', 'BY', 'BZ', 'CA', 'CC', 'CD', 'CF', 'CG', 'CH', 'CI', 'CK', 'CL', 'CM', 'CN', 'CO', 'CR', 'CU', 'CV', 'CW', 'CX', 'CY', 'CZ', 'DE', 'DJ', 'DK', 'DM', 'DO', 'DZ', 'EC', 'EE', 'EG', 'EH', 'ER', 'ES', 'ET', 'FI', 'FJ', 'FK', 'FM', 'FO', 'FR', 'GA', 'GB', 'GD', 'GE', 'GF', 'GG', 'GH', 'GI', 'GL', 'GM', 'GN', 'GP', 'GQ', 'GR', 'GS', 'GT', 'GU', 'GW', 'GY', 'HK', 'HN', 'HR', 'HT', 'HU', 'ID', 'IE', 'IL', 'IM', 'IN', 'IO', 'IQ', 'IR', 'IS', 'IT', 'JE', 'JM', 'JO', 'JP', 'KE', 'KG', 'KH', 'KI', 'KM', 'KN', 'KP', 'KR', 'KW', 'KY', 'KZ', 'LA', 'LB', 'LC', 'LI', 'LK', 'LR', 'LS', 'LT', 'LU', 'LV', 'LY', 'MA', 'MC', 'MD', 'ME', 'MF', 'MG', 'MH', 'MK', 'ML', 'MM', 'MN', 'MO', 'MP', 'MQ', 'MR', 'MS', 'MT', 'MU', 'MV', 'MW', 'MX', 'MY', 'MZ', 'NA', 'NC', 'NE', 'NF', 'NG', 'NI', 'NL', 'NO', 'NP', 'NR', 'NU', 'NZ', 'OM', 'PA', 'PE', 'PF', 'PG', 'PH', 'PK', 'PL', 'PM', 'PN', 'PR', 'PS', 'PT', 'PW', 'PY', 'QA', 'RE', 'RO', 'RS', 'RU', 'RW', 'SA', 'SB', 'SC', 'SD', 'SE', 'SG', 'SH', 'SI', 'SJ', 'SK', 'SL', 'SM', 'SN', 'SO', 'SR', 'SS', 'ST', 'SV', 'SX', 'SY', 'SZ', 'TC', 'TD', 'TG', 'TH', 'TJ', 'TK', 'TL', 'TM', 'TN', 'TO', 'TR', 'TT', 'TV', 'TW', 'TZ', 'UA', 'UG', 'UM', 'US', 'UY', 'UZ', 'VA', 'VC', 'VE', 'VG', 'VI', 'VN', 'VU', 'WF', 'WS', 'XK', 'YE', 'YT', 'ZA', 'ZM', 'ZW' */
export var Country;
(function (Country) {
    Country["AD"] = "AD";
    Country["AE"] = "AE";
    Country["AF"] = "AF";
    Country["AG"] = "AG";
    Country["AI"] = "AI";
    Country["AL"] = "AL";
    Country["AM"] = "AM";
    Country["AN"] = "AN";
    Country["AO"] = "AO";
    Country["AR"] = "AR";
    Country["AS"] = "AS";
    Country["AT"] = "AT";
    Country["AU"] = "AU";
    Country["AW"] = "AW";
    Country["AX"] = "AX";
    Country["AZ"] = "AZ";
    Country["BA"] = "BA";
    Country["BB"] = "BB";
    Country["BD"] = "BD";
    Country["BE"] = "BE";
    Country["BF"] = "BF";
    Country["BG"] = "BG";
    Country["BH"] = "BH";
    Country["BI"] = "BI";
    Country["BJ"] = "BJ";
    Country["BL"] = "BL";
    Country["BM"] = "BM";
    Country["BN"] = "BN";
    Country["BO"] = "BO";
    Country["BQ"] = "BQ";
    Country["BR"] = "BR";
    Country["BS"] = "BS";
    Country["BT"] = "BT";
    Country["BW"] = "BW";
    Country["BY"] = "BY";
    Country["BZ"] = "BZ";
    Country["CA"] = "CA";
    Country["CC"] = "CC";
    Country["CD"] = "CD";
    Country["CF"] = "CF";
    Country["CG"] = "CG";
    Country["CH"] = "CH";
    Country["CI"] = "CI";
    Country["CK"] = "CK";
    Country["CL"] = "CL";
    Country["CM"] = "CM";
    Country["CN"] = "CN";
    Country["CO"] = "CO";
    Country["CR"] = "CR";
    Country["CU"] = "CU";
    Country["CV"] = "CV";
    Country["CW"] = "CW";
    Country["CX"] = "CX";
    Country["CY"] = "CY";
    Country["CZ"] = "CZ";
    Country["DE"] = "DE";
    Country["DJ"] = "DJ";
    Country["DK"] = "DK";
    Country["DM"] = "DM";
    Country["DO"] = "DO";
    Country["DZ"] = "DZ";
    Country["EC"] = "EC";
    Country["EE"] = "EE";
    Country["EG"] = "EG";
    Country["EH"] = "EH";
    Country["ER"] = "ER";
    Country["ES"] = "ES";
    Country["ET"] = "ET";
    Country["FI"] = "FI";
    Country["FJ"] = "FJ";
    Country["FK"] = "FK";
    Country["FM"] = "FM";
    Country["FO"] = "FO";
    Country["FR"] = "FR";
    Country["GA"] = "GA";
    Country["GB"] = "GB";
    Country["GD"] = "GD";
    Country["GE"] = "GE";
    Country["GF"] = "GF";
    Country["GG"] = "GG";
    Country["GH"] = "GH";
    Country["GI"] = "GI";
    Country["GL"] = "GL";
    Country["GM"] = "GM";
    Country["GN"] = "GN";
    Country["GP"] = "GP";
    Country["GQ"] = "GQ";
    Country["GR"] = "GR";
    Country["GS"] = "GS";
    Country["GT"] = "GT";
    Country["GU"] = "GU";
    Country["GW"] = "GW";
    Country["GY"] = "GY";
    Country["HK"] = "HK";
    Country["HN"] = "HN";
    Country["HR"] = "HR";
    Country["HT"] = "HT";
    Country["HU"] = "HU";
    Country["ID"] = "ID";
    Country["IE"] = "IE";
    Country["IL"] = "IL";
    Country["IM"] = "IM";
    Country["IN"] = "IN";
    Country["IO"] = "IO";
    Country["IQ"] = "IQ";
    Country["IR"] = "IR";
    Country["IS"] = "IS";
    Country["IT"] = "IT";
    Country["JE"] = "JE";
    Country["JM"] = "JM";
    Country["JO"] = "JO";
    Country["JP"] = "JP";
    Country["KE"] = "KE";
    Country["KG"] = "KG";
    Country["KH"] = "KH";
    Country["KI"] = "KI";
    Country["KM"] = "KM";
    Country["KN"] = "KN";
    Country["KP"] = "KP";
    Country["KR"] = "KR";
    Country["KW"] = "KW";
    Country["KY"] = "KY";
    Country["KZ"] = "KZ";
    Country["LA"] = "LA";
    Country["LB"] = "LB";
    Country["LC"] = "LC";
    Country["LI"] = "LI";
    Country["LK"] = "LK";
    Country["LR"] = "LR";
    Country["LS"] = "LS";
    Country["LT"] = "LT";
    Country["LU"] = "LU";
    Country["LV"] = "LV";
    Country["LY"] = "LY";
    Country["MA"] = "MA";
    Country["MC"] = "MC";
    Country["MD"] = "MD";
    Country["ME"] = "ME";
    Country["MF"] = "MF";
    Country["MG"] = "MG";
    Country["MH"] = "MH";
    Country["MK"] = "MK";
    Country["ML"] = "ML";
    Country["MM"] = "MM";
    Country["MN"] = "MN";
    Country["MO"] = "MO";
    Country["MP"] = "MP";
    Country["MQ"] = "MQ";
    Country["MR"] = "MR";
    Country["MS"] = "MS";
    Country["MT"] = "MT";
    Country["MU"] = "MU";
    Country["MV"] = "MV";
    Country["MW"] = "MW";
    Country["MX"] = "MX";
    Country["MY"] = "MY";
    Country["MZ"] = "MZ";
    Country["NA"] = "NA";
    Country["NC"] = "NC";
    Country["NE"] = "NE";
    Country["NF"] = "NF";
    Country["NG"] = "NG";
    Country["NI"] = "NI";
    Country["NL"] = "NL";
    Country["NO"] = "NO";
    Country["NP"] = "NP";
    Country["NR"] = "NR";
    Country["NU"] = "NU";
    Country["NZ"] = "NZ";
    Country["OM"] = "OM";
    Country["PA"] = "PA";
    Country["PE"] = "PE";
    Country["PF"] = "PF";
    Country["PG"] = "PG";
    Country["PH"] = "PH";
    Country["PK"] = "PK";
    Country["PL"] = "PL";
    Country["PM"] = "PM";
    Country["PN"] = "PN";
    Country["PR"] = "PR";
    Country["PS"] = "PS";
    Country["PT"] = "PT";
    Country["PW"] = "PW";
    Country["PY"] = "PY";
    Country["QA"] = "QA";
    Country["RE"] = "RE";
    Country["RO"] = "RO";
    Country["RS"] = "RS";
    Country["RU"] = "RU";
    Country["RW"] = "RW";
    Country["SA"] = "SA";
    Country["SB"] = "SB";
    Country["SC"] = "SC";
    Country["SD"] = "SD";
    Country["SE"] = "SE";
    Country["SG"] = "SG";
    Country["SH"] = "SH";
    Country["SI"] = "SI";
    Country["SJ"] = "SJ";
    Country["SK"] = "SK";
    Country["SL"] = "SL";
    Country["SM"] = "SM";
    Country["SN"] = "SN";
    Country["SO"] = "SO";
    Country["SR"] = "SR";
    Country["SS"] = "SS";
    Country["ST"] = "ST";
    Country["SV"] = "SV";
    Country["SX"] = "SX";
    Country["SY"] = "SY";
    Country["SZ"] = "SZ";
    Country["TC"] = "TC";
    Country["TD"] = "TD";
    Country["TG"] = "TG";
    Country["TH"] = "TH";
    Country["TJ"] = "TJ";
    Country["TK"] = "TK";
    Country["TL"] = "TL";
    Country["TM"] = "TM";
    Country["TN"] = "TN";
    Country["TO"] = "TO";
    Country["TR"] = "TR";
    Country["TT"] = "TT";
    Country["TV"] = "TV";
    Country["TW"] = "TW";
    Country["TZ"] = "TZ";
    Country["UA"] = "UA";
    Country["UG"] = "UG";
    Country["UM"] = "UM";
    Country["US"] = "US";
    Country["UY"] = "UY";
    Country["UZ"] = "UZ";
    Country["VA"] = "VA";
    Country["VC"] = "VC";
    Country["VE"] = "VE";
    Country["VG"] = "VG";
    Country["VI"] = "VI";
    Country["VN"] = "VN";
    Country["VU"] = "VU";
    Country["WF"] = "WF";
    Country["WS"] = "WS";
    Country["XK"] = "XK";
    Country["YE"] = "YE";
    Country["YT"] = "YT";
    Country["ZA"] = "ZA";
    Country["ZM"] = "ZM";
    Country["ZW"] = "ZW";
})(Country || (Country = {}));
/** One of 'AED', 'AFN', 'ALL', 'AMD', 'AOA', 'ARS', 'AUD', 'AWG', 'AZN', 'BAM', 'BBD', 'BDT', 'BGN', 'BHD', 'BIF', 'BMD', 'BND', 'BOB', 'BOV', 'BRL', 'BSD', 'BTC', 'BTN', 'BWP', 'BYN', 'BZD', 'CAD', 'CDF', 'CHE', 'CHF', 'CHW', 'CLF', 'CLP', 'CNY', 'COP', 'COU', 'CRC', 'CUP', 'CVE', 'CZK', 'DJF', 'DKK', 'DOP', 'DZD', 'EGP', 'ERN', 'ETB', 'ETH', 'EUR', 'FJD', 'FKP', 'GBP', 'GEL', 'GHS', 'GIP', 'GMD', 'GNF', 'GTQ', 'GYD', 'HKD', 'HNL', 'HTG', 'HUF', 'IDR', 'ILS', 'INR', 'IQD', 'IRR', 'ISK', 'JMD', 'JOD', 'JPY', 'KES', 'KGS', 'KHR', 'KMF', 'KPW', 'KRW', 'KWD', 'KYD', 'KZT', 'LAK', 'LBP', 'LKR', 'LRD', 'LSL', 'LYD', 'MAD', 'MDL', 'MGA', 'MKD', 'MMK', 'MNT', 'MOP', 'MRU', 'MUR', 'MVR', 'MWK', 'MXN', 'MXV', 'MYR', 'MZN', 'NAD', 'NGN', 'NIO', 'NOK', 'NPR', 'NZD', 'OMR', 'PAB', 'PEN', 'PGK', 'PHP', 'PKR', 'PLN', 'PYG', 'QAR', 'RON', 'RSD', 'RUB', 'RWF', 'SAR', 'SBD', 'SCR', 'SDG', 'SEK', 'SGD', 'SHP', 'SLE', 'SOS', 'SRD', 'SSP', 'STN', 'SVC', 'SYP', 'SZL', 'THB', 'TJS', 'TMT', 'TND', 'TOP', 'TRY', 'TTD', 'TWD', 'TZS', 'UAH', 'UGX', 'USD', 'USN', 'UYI', 'UYU', 'UYW', 'UZS', 'VED', 'VES', 'VND', 'VUV', 'WST', 'XAD', 'XAF', 'XAG', 'XAU', 'XBA', 'XBB', 'XBC', 'XBD', 'XCD', 'XCG', 'XDR', 'XOF', 'XPD', 'XPF', 'XPT', 'XSU', 'XTS', 'XUA', 'XXX', 'YER', 'ZAR', 'ZMW', 'ZWG' */
export var Currency;
(function (Currency) {
    Currency["AED"] = "AED";
    Currency["AFN"] = "AFN";
    Currency["ALL"] = "ALL";
    Currency["AMD"] = "AMD";
    Currency["AOA"] = "AOA";
    Currency["ARS"] = "ARS";
    Currency["AUD"] = "AUD";
    Currency["AWG"] = "AWG";
    Currency["AZN"] = "AZN";
    Currency["BAM"] = "BAM";
    Currency["BBD"] = "BBD";
    Currency["BDT"] = "BDT";
    Currency["BGN"] = "BGN";
    Currency["BHD"] = "BHD";
    Currency["BIF"] = "BIF";
    Currency["BMD"] = "BMD";
    Currency["BND"] = "BND";
    Currency["BOB"] = "BOB";
    Currency["BOV"] = "BOV";
    Currency["BRL"] = "BRL";
    Currency["BSD"] = "BSD";
    Currency["BTC"] = "BTC";
    Currency["BTN"] = "BTN";
    Currency["BWP"] = "BWP";
    Currency["BYN"] = "BYN";
    Currency["BZD"] = "BZD";
    Currency["CAD"] = "CAD";
    Currency["CDF"] = "CDF";
    Currency["CHE"] = "CHE";
    Currency["CHF"] = "CHF";
    Currency["CHW"] = "CHW";
    Currency["CLF"] = "CLF";
    Currency["CLP"] = "CLP";
    Currency["CNY"] = "CNY";
    Currency["COP"] = "COP";
    Currency["COU"] = "COU";
    Currency["CRC"] = "CRC";
    Currency["CUP"] = "CUP";
    Currency["CVE"] = "CVE";
    Currency["CZK"] = "CZK";
    Currency["DJF"] = "DJF";
    Currency["DKK"] = "DKK";
    Currency["DOP"] = "DOP";
    Currency["DZD"] = "DZD";
    Currency["EGP"] = "EGP";
    Currency["ERN"] = "ERN";
    Currency["ETB"] = "ETB";
    Currency["ETH"] = "ETH";
    Currency["EUR"] = "EUR";
    Currency["FJD"] = "FJD";
    Currency["FKP"] = "FKP";
    Currency["GBP"] = "GBP";
    Currency["GEL"] = "GEL";
    Currency["GHS"] = "GHS";
    Currency["GIP"] = "GIP";
    Currency["GMD"] = "GMD";
    Currency["GNF"] = "GNF";
    Currency["GTQ"] = "GTQ";
    Currency["GYD"] = "GYD";
    Currency["HKD"] = "HKD";
    Currency["HNL"] = "HNL";
    Currency["HTG"] = "HTG";
    Currency["HUF"] = "HUF";
    Currency["IDR"] = "IDR";
    Currency["ILS"] = "ILS";
    Currency["INR"] = "INR";
    Currency["IQD"] = "IQD";
    Currency["IRR"] = "IRR";
    Currency["ISK"] = "ISK";
    Currency["JMD"] = "JMD";
    Currency["JOD"] = "JOD";
    Currency["JPY"] = "JPY";
    Currency["KES"] = "KES";
    Currency["KGS"] = "KGS";
    Currency["KHR"] = "KHR";
    Currency["KMF"] = "KMF";
    Currency["KPW"] = "KPW";
    Currency["KRW"] = "KRW";
    Currency["KWD"] = "KWD";
    Currency["KYD"] = "KYD";
    Currency["KZT"] = "KZT";
    Currency["LAK"] = "LAK";
    Currency["LBP"] = "LBP";
    Currency["LKR"] = "LKR";
    Currency["LRD"] = "LRD";
    Currency["LSL"] = "LSL";
    Currency["LYD"] = "LYD";
    Currency["MAD"] = "MAD";
    Currency["MDL"] = "MDL";
    Currency["MGA"] = "MGA";
    Currency["MKD"] = "MKD";
    Currency["MMK"] = "MMK";
    Currency["MNT"] = "MNT";
    Currency["MOP"] = "MOP";
    Currency["MRU"] = "MRU";
    Currency["MUR"] = "MUR";
    Currency["MVR"] = "MVR";
    Currency["MWK"] = "MWK";
    Currency["MXN"] = "MXN";
    Currency["MXV"] = "MXV";
    Currency["MYR"] = "MYR";
    Currency["MZN"] = "MZN";
    Currency["NAD"] = "NAD";
    Currency["NGN"] = "NGN";
    Currency["NIO"] = "NIO";
    Currency["NOK"] = "NOK";
    Currency["NPR"] = "NPR";
    Currency["NZD"] = "NZD";
    Currency["OMR"] = "OMR";
    Currency["PAB"] = "PAB";
    Currency["PEN"] = "PEN";
    Currency["PGK"] = "PGK";
    Currency["PHP"] = "PHP";
    Currency["PKR"] = "PKR";
    Currency["PLN"] = "PLN";
    Currency["PYG"] = "PYG";
    Currency["QAR"] = "QAR";
    Currency["RON"] = "RON";
    Currency["RSD"] = "RSD";
    Currency["RUB"] = "RUB";
    Currency["RWF"] = "RWF";
    Currency["SAR"] = "SAR";
    Currency["SBD"] = "SBD";
    Currency["SCR"] = "SCR";
    Currency["SDG"] = "SDG";
    Currency["SEK"] = "SEK";
    Currency["SGD"] = "SGD";
    Currency["SHP"] = "SHP";
    Currency["SLE"] = "SLE";
    Currency["SOS"] = "SOS";
    Currency["SRD"] = "SRD";
    Currency["SSP"] = "SSP";
    Currency["STN"] = "STN";
    Currency["SVC"] = "SVC";
    Currency["SYP"] = "SYP";
    Currency["SZL"] = "SZL";
    Currency["THB"] = "THB";
    Currency["TJS"] = "TJS";
    Currency["TMT"] = "TMT";
    Currency["TND"] = "TND";
    Currency["TOP"] = "TOP";
    Currency["TRY"] = "TRY";
    Currency["TTD"] = "TTD";
    Currency["TWD"] = "TWD";
    Currency["TZS"] = "TZS";
    Currency["UAH"] = "UAH";
    Currency["UGX"] = "UGX";
    Currency["USD"] = "USD";
    Currency["USN"] = "USN";
    Currency["UYI"] = "UYI";
    Currency["UYU"] = "UYU";
    Currency["UYW"] = "UYW";
    Currency["UZS"] = "UZS";
    Currency["VED"] = "VED";
    Currency["VES"] = "VES";
    Currency["VND"] = "VND";
    Currency["VUV"] = "VUV";
    Currency["WST"] = "WST";
    Currency["XAD"] = "XAD";
    Currency["XAF"] = "XAF";
    Currency["XAG"] = "XAG";
    Currency["XAU"] = "XAU";
    Currency["XBA"] = "XBA";
    Currency["XBB"] = "XBB";
    Currency["XBC"] = "XBC";
    Currency["XBD"] = "XBD";
    Currency["XCD"] = "XCD";
    Currency["XCG"] = "XCG";
    Currency["XDR"] = "XDR";
    Currency["XOF"] = "XOF";
    Currency["XPD"] = "XPD";
    Currency["XPF"] = "XPF";
    Currency["XPT"] = "XPT";
    Currency["XSU"] = "XSU";
    Currency["XTS"] = "XTS";
    Currency["XUA"] = "XUA";
    Currency["XXX"] = "XXX";
    Currency["YER"] = "YER";
    Currency["ZAR"] = "ZAR";
    Currency["ZMW"] = "ZMW";
    Currency["ZWG"] = "ZWG";
})(Currency || (Currency = {}));
/** One of 'bool', 'date', 'text' */
export var CustomFieldType;
(function (CustomFieldType) {
    CustomFieldType["Bool"] = "bool";
    CustomFieldType["Date"] = "date";
    CustomFieldType["Text"] = "text";
})(CustomFieldType || (CustomFieldType = {}));
/** One of 'dmy_dashes', 'dmy_slashes', 'mdy_dashes', 'mdy_slashes', 'ymd_dashes', 'ymd_slashes' */
export var DateFormat;
(function (DateFormat) {
    DateFormat["Dmy_dashes"] = "dmy_dashes";
    DateFormat["Dmy_slashes"] = "dmy_slashes";
    DateFormat["Mdy_dashes"] = "mdy_dashes";
    DateFormat["Mdy_slashes"] = "mdy_slashes";
    DateFormat["Ymd_dashes"] = "ymd_dashes";
    DateFormat["Ymd_slashes"] = "ymd_slashes";
})(DateFormat || (DateFormat = {}));
/** One of 'feedbacks', 'scheduledGiving', 'sponsorships' */
export var Feature;
(function (Feature) {
    Feature["Feedbacks"] = "feedbacks";
    Feature["ScheduledGiving"] = "scheduledGiving";
    Feature["Sponsorships"] = "sponsorships";
})(Feature || (Feature = {}));
/** One of 'dropdown', 'toggle' */
export var FundDimensionSelector;
(function (FundDimensionSelector) {
    FundDimensionSelector["Dropdown"] = "dropdown";
    FundDimensionSelector["Toggle"] = "toggle";
})(FundDimensionSelector || (FundDimensionSelector = {}));
/** One of 'oneTime', 'recurring' */
export var GiftType;
(function (GiftType) {
    GiftType["OneTime"] = "oneTime";
    GiftType["Recurring"] = "recurring";
})(GiftType || (GiftType = {}));
/** One of 'countries', 'currencies' */
export var IconCollection;
(function (IconCollection) {
    IconCollection["Countries"] = "countries";
    IconCollection["Currencies"] = "currencies";
})(IconCollection || (IconCollection = {}));
export var IsoDayOfWeek;
(function (IsoDayOfWeek) {
    IsoDayOfWeek[IsoDayOfWeek["_0"] = 0] = "_0";
    IsoDayOfWeek[IsoDayOfWeek["_1"] = 1] = "_1";
    IsoDayOfWeek[IsoDayOfWeek["_2"] = 2] = "_2";
    IsoDayOfWeek[IsoDayOfWeek["_3"] = 3] = "_3";
    IsoDayOfWeek[IsoDayOfWeek["_4"] = 4] = "_4";
    IsoDayOfWeek[IsoDayOfWeek["_5"] = 5] = "_5";
    IsoDayOfWeek[IsoDayOfWeek["_6"] = 6] = "_6";
    IsoDayOfWeek[IsoDayOfWeek["_7"] = 7] = "_7";
})(IsoDayOfWeek || (IsoDayOfWeek = {}));
/** One of 'logo', 'logoDark' */
export var MobileAppAssetType;
(function (MobileAppAssetType) {
    MobileAppAssetType["Logo"] = "logo";
    MobileAppAssetType["LogoDark"] = "logoDark";
})(MobileAppAssetType || (MobileAppAssetType = {}));
/** One of 'layout1', 'layout2' */
export var NameLayout;
(function (NameLayout) {
    NameLayout["Layout1"] = "layout1";
    NameLayout["Layout2"] = "layout2";
})(NameLayout || (NameLayout = {}));
/** One of 'eu1', 'eu2', 'international' */
export var NumberFormat;
(function (NumberFormat) {
    NumberFormat["Eu1"] = "eu1";
    NumberFormat["Eu2"] = "eu2";
    NumberFormat["International"] = "international";
})(NumberFormat || (NumberFormat = {}));
/** One of 'cancelled', 'declined', 'draft', 'error', 'paid', 'processing', 'refunded', 'returned' */
export var PaymentStatus;
(function (PaymentStatus) {
    PaymentStatus["Cancelled"] = "cancelled";
    PaymentStatus["Declined"] = "declined";
    PaymentStatus["Draft"] = "draft";
    PaymentStatus["Error"] = "error";
    PaymentStatus["Paid"] = "paid";
    PaymentStatus["Processing"] = "processing";
    PaymentStatus["Refunded"] = "refunded";
    PaymentStatus["Returned"] = "returned";
})(PaymentStatus || (PaymentStatus = {}));
/** One of 'noResponse', 'optIn', 'optOut' */
export var Preference;
(function (Preference) {
    Preference["NoResponse"] = "noResponse";
    Preference["OptIn"] = "optIn";
    Preference["OptOut"] = "optOut";
})(Preference || (Preference = {}));
/** One of 'type1', 'type2' */
export var ProcessorType;
(function (ProcessorType) {
    ProcessorType["Type1"] = "type1";
    ProcessorType["Type2"] = "type2";
})(ProcessorType || (ProcessorType = {}));
/** One of 'build', 'campaign', 'designation', 'element', 'sponsorships', 'subscription' */
export var PublishedFileKind;
(function (PublishedFileKind) {
    PublishedFileKind["Build"] = "build";
    PublishedFileKind["Campaign"] = "campaign";
    PublishedFileKind["Designation"] = "designation";
    PublishedFileKind["Element"] = "element";
    PublishedFileKind["Sponsorships"] = "sponsorships";
    PublishedFileKind["Subscription"] = "subscription";
})(PublishedFileKind || (PublishedFileKind = {}));
/** One of 'local', 'prod', 'qa' */
export var ServiceEnvironment;
(function (ServiceEnvironment) {
    ServiceEnvironment["Local"] = "local";
    ServiceEnvironment["Prod"] = "prod";
    ServiceEnvironment["Qa"] = "qa";
})(ServiceEnvironment || (ServiceEnvironment = {}));
/** One of 'error', 'warning' */
export var Severity;
(function (Severity) {
    Severity["Error"] = "error";
    Severity["Warning"] = "warning";
})(Severity || (Severity = {}));
/** One of 'accounts.organizationType', 'accounts.title', 'communications.communicationCategory', 'communications.communicationChannel', 'subscriptions.country' */
export var SubscriptionLookupType;
(function (SubscriptionLookupType) {
    SubscriptionLookupType["Accounts_organizationType"] = "accounts.organizationType";
    SubscriptionLookupType["Accounts_title"] = "accounts.title";
    SubscriptionLookupType["Communications_communicationCategory"] = "communications.communicationCategory";
    SubscriptionLookupType["Communications_communicationChannel"] = "communications.communicationChannel";
    SubscriptionLookupType["Subscriptions_country"] = "subscriptions.country";
})(SubscriptionLookupType || (SubscriptionLookupType = {}));
/** One of 'en', 'es', 'fr', 'xx' */
export var SystemLanguage;
(function (SystemLanguage) {
    SystemLanguage["En"] = "en";
    SystemLanguage["Es"] = "es";
    SystemLanguage["Fr"] = "fr";
    SystemLanguage["Xx"] = "xx";
})(SystemLanguage || (SystemLanguage = {}));
/** One of 'crowdfunding::crowdfunding-page::crowdfunder-page-name', 'crowdfunding::crowdfunding-page::crowdfunder-page-url', 'general::fundraising::campaign', 'general::fundraising::fundraiser', 'platforms::browser::engine', 'platforms::browser::major-version', 'platforms::browser::name', 'platforms::browser::type', 'platforms::browser::user-agent', 'platforms::browser::version', 'platforms::device::cpu-architecture', 'platforms::device::model', 'platforms::device::operating-system', 'platforms::device::operating-system-version', 'platforms::device::type', 'platforms::device::vendor', 'platforms::donation-form::designation', 'platforms::donation-form::suggested-amount-used', 'platforms::donation-form::url', 'platforms::session::duration-seconds', 'platforms::session::landing-url', 'web::email::content', 'web::email::list', 'web::email::segment', 'web::google-analytics::campaign', 'web::google-analytics::content', 'web::google-analytics::medium', 'web::google-analytics::source', 'web::google-analytics::term' */
export var SystemTag;
(function (SystemTag) {
    SystemTag["CrowdfundingCrowdfundingPageCrowdfunderPageName"] = "crowdfunding::crowdfunding-page::crowdfunder-page-name";
    SystemTag["CrowdfundingCrowdfundingPageCrowdfunderPageUrl"] = "crowdfunding::crowdfunding-page::crowdfunder-page-url";
    SystemTag["GeneralFundraisingCampaign"] = "general::fundraising::campaign";
    SystemTag["GeneralFundraisingFundraiser"] = "general::fundraising::fundraiser";
    SystemTag["PlatformsBrowserEngine"] = "platforms::browser::engine";
    SystemTag["PlatformsBrowserMajorVersion"] = "platforms::browser::major-version";
    SystemTag["PlatformsBrowserName"] = "platforms::browser::name";
    SystemTag["PlatformsBrowserType"] = "platforms::browser::type";
    SystemTag["PlatformsBrowserUserAgent"] = "platforms::browser::user-agent";
    SystemTag["PlatformsBrowserVersion"] = "platforms::browser::version";
    SystemTag["PlatformsDeviceCpuArchitecture"] = "platforms::device::cpu-architecture";
    SystemTag["PlatformsDeviceModel"] = "platforms::device::model";
    SystemTag["PlatformsDeviceOperatingSystem"] = "platforms::device::operating-system";
    SystemTag["PlatformsDeviceOperatingSystemVersion"] = "platforms::device::operating-system-version";
    SystemTag["PlatformsDeviceType"] = "platforms::device::type";
    SystemTag["PlatformsDeviceVendor"] = "platforms::device::vendor";
    SystemTag["PlatformsDonationFormDesignation"] = "platforms::donation-form::designation";
    SystemTag["PlatformsDonationFormSuggestedAmountUsed"] = "platforms::donation-form::suggested-amount-used";
    SystemTag["PlatformsDonationFormUrl"] = "platforms::donation-form::url";
    SystemTag["PlatformsSessionDurationSeconds"] = "platforms::session::duration-seconds";
    SystemTag["PlatformsSessionLandingUrl"] = "platforms::session::landing-url";
    SystemTag["WebEmailContent"] = "web::email::content";
    SystemTag["WebEmailList"] = "web::email::list";
    SystemTag["WebEmailSegment"] = "web::email::segment";
    SystemTag["WebGoogleAnalyticsCampaign"] = "web::google-analytics::campaign";
    SystemTag["WebGoogleAnalyticsContent"] = "web::google-analytics::content";
    SystemTag["WebGoogleAnalyticsMedium"] = "web::google-analytics::medium";
    SystemTag["WebGoogleAnalyticsSource"] = "web::google-analytics::source";
    SystemTag["WebGoogleAnalyticsTerm"] = "web::google-analytics::term";
})(SystemTag || (SystemTag = {}));
/** One of 'crowdfunding', 'general', 'platforms', 'web' */
export var SystemTagCategory;
(function (SystemTagCategory) {
    SystemTagCategory["Crowdfunding"] = "crowdfunding";
    SystemTagCategory["General"] = "general";
    SystemTagCategory["Platforms"] = "platforms";
    SystemTagCategory["Web"] = "web";
})(SystemTagCategory || (SystemTagCategory = {}));
/** One of 'crowdfunding_crowdfundingPage', 'general_fundraising', 'platforms_browser', 'platforms_device', 'platforms_donationForm', 'platforms_session', 'web_email', 'web_googleAnalytics' */
export var SystemTagGroup;
(function (SystemTagGroup) {
    SystemTagGroup["Crowdfunding_crowdfundingPage"] = "crowdfunding_crowdfundingPage";
    SystemTagGroup["General_fundraising"] = "general_fundraising";
    SystemTagGroup["Platforms_browser"] = "platforms_browser";
    SystemTagGroup["Platforms_device"] = "platforms_device";
    SystemTagGroup["Platforms_donationForm"] = "platforms_donationForm";
    SystemTagGroup["Platforms_session"] = "platforms_session";
    SystemTagGroup["Web_email"] = "web_email";
    SystemTagGroup["Web_googleAnalytics"] = "web_googleAnalytics";
})(SystemTagGroup || (SystemTagGroup = {}));
/** One of 'lowercase', 'titlecase', 'uppercase' */
export var TextTransformation;
(function (TextTransformation) {
    TextTransformation["Lowercase"] = "lowercase";
    TextTransformation["Titlecase"] = "titlecase";
    TextTransformation["Uppercase"] = "uppercase";
})(TextTransformation || (TextTransformation = {}));
/** One of '12', '24' */
export var TimeFormat;
(function (TimeFormat) {
    TimeFormat["_12"] = "12";
    TimeFormat["_24"] = "24";
})(TimeFormat || (TimeFormat = {}));
